test/component_activated_menu_test.js

```diff
@@ -0,0 +1,322 @@
+      preventDefault: true,
+      selection_event: TEST_SELECTION_EVENT_NAME
+    });
+  });
+
+  after(function() {
+    menu.$node.menu("destroy");
+    menu.$node.remove();
+    menu.activator.$node.remove();
+    menu.container.$node.remove();
+    menu = null;
+  });
+
+  describe("ActivatedMenu", function() {
+    it("should have the basic HTML in place", function() {
+      expect($("#" + MENU_ID).length).to.equal(1);
+    });
+
+    it("should have the component class name present", function() {
+      expect($("#" + MENU_ID).parent().hasClass(COMPONENT_CLASSNAME)).to.be.true;
+    });
+
+    it("should make the $node public", function() {
+      expect(menu.$node).to.exist;
+      expect(menu.$node.length).to.equal(1);
+    });
+
+    it("should make the activator public", function() {
+      expect(menu.activator).to.exist;
+    });
+
+    it("should make the container public", function() {
+      expect(menu.container).to.exist;
+    });
+
+    it("should make the instance available as data on the $node", function() {
+      expect(menu.$node.data("instance")).to.exist;
+      expect(menu.$node.data("instance")).to.equal(menu);
+    });
+
+    it("should make (public but indicated as) private reference to config", function() {
+      expect(menu._config).to.exist;
+      expect(menu._config.preventDefault).to.exist;
+      expect(menu._config.activator_text).to.equal(ACTIVATOR_TEXT);
+    });
+
+    it("should make (public but indicated as) private reference to position", function() {
+      expect(menu._position).to.exist;
+      expect(menu._position.my).to.exist;
+      expect(menu._position.my).to.equal("left top");
+      expect(menu._position.at).to.exist;
+      expect(menu._position.at).to.equal("left bottom");
+      expect(menu._position.of).to.exist;
+    });
+
+    it("should make (public but indicated as) private reference to state", function() {
+      expect(menu._state).to.exist;
+      expect(menu._state.open).to.exist;
+    });
+
+    it("should open the menu by the open() method", function() {
+      simulateClosed(menu);
+      expect(menu.container.$node.get(0).style.display).to.equal("none");
+
+      menu.open();
+      expect(menu.container.$node.get(0).style.display).to.equal("");
+    });
+
+    it("should set the state.open to true when open() is activated", function() {
+      simulateClosed(menu);
+      expect(menu._state.open).to.be.false;
+
+      menu.open();
+      expect(menu._state.open).to.be.true;
+    });
+
+    it("should add the class 'active' to the activator on open()", function() {
+      simulateClosed(menu);
+
+      expect(menu._state.open).to.be.false;
+      expect(menu.activator.$node.hasClass("active")).to.be.false;
+
+      menu.open();
+      expect(menu.activator.$node.hasClass("active")).to.be.true;
+    });
+
+    /* TODO: Test is on hold because we cannot use jsDom
+     * for testing position of elements
+     **/
+    it("should set the position values passed in open()");
```

Comment:
are we going to circle back round to do this? if we may as well remove it

Reply:
Yes, we should add this test.

Reason I haven't, yet, is because the current framework, specifically jsDom, does not support position values. I have not found a suitable (small/simple/minimal effort) alternative yet.

Leaving in this state is what Mocha calls a 'pending' test. It does not produce a fail but does leave a constant reminder, for anyone looking at output, that it needs finishing.
https://mochajs.org/#pending-tests